src/App.tsx

```diff
@@ -1,204 +1,110 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { ErrorMessages, FilterStatus, Todo } from './types/Todo';
+import { Header } from './component/Header';
+import { Footer } from './component/Footer';
+import { ErrorNotification } from './component/ErrorNotification';
+import { TodoList } from './component/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages>(
+    ErrorMessages.DEFAULT,
+  );
+  const [loading, setLoading] = useState(true);
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(
+    FilterStatus.ALL,
+  );
+
+  function loadTodos() {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LOAD_TODOS);
+        setTimeout(() => {
+          setErrorMessage(ErrorMessages.DEFAULT);
+        }, 3000);
+      })
+      .finally(() => setLoading(false));
   }
 
+  useEffect(loadTodos, []);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case FilterStatus.ACTIVE:
+        return !todo.completed;
+      case FilterStatus.COMPLETED:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const handleDeleteTodo = (todoId: number) => {
+    setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
```

Comment:
```suggestion
  const handleDeleteTodo = (todoId: number) => {
    const selectedTodo = currentTodos.filter(todo => todo.id !== todoId);
    setTodos(selectedTodo);
```
bad practice use functions as args, create const instead